src/include/storage/buffer_manager/page_state.h

```diff
@@ -74,12 +81,28 @@ class PageState {
     bool isDirty() const { return stateAndVersion & DIRTY_MASK; }
     uint64_t getStateAndVersion() const { return stateAndVersion.load(); }
 
-    void resetToEvicted() { stateAndVersion.store(EVICTED << NUM_BITS_TO_SHIFT_FOR_STATE); }
+    void resetToEvicted() {
+        stateAndVersion.store(EVICTED << NUM_BITS_TO_SHIFT_FOR_STATE);
+#if BM_MALLOC
+        page.reset();
+#endif
+    }
+
+#if BM_MALLOC
+    uint8_t* getPage() const { return page.get(); }
+    uint8_t* allocatePage(uint64_t pageSize) {
+        page = std::make_unique<uint8_t[]>(pageSize);
```

Comment:
Interestingly in webassembly the tests are compiled with the default [ABORTING_MALLOC=1](https://emscripten.org/docs/tools_reference/settings_reference.html?highlight=environment#aborting-malloc), which aborts the webassembly process if it runs out of memory instead of returning 0, but outside of the tests we're compiling with `ALLOW_MEMORY_GROWTH`, which has the default of having malloc return 0 on failure.
So I think we should set `ABORTING_MALLOC=0` for the tests just so it's handled the same (not that we have any tests where we expect this yet).